packages/snaps-controllers/src/snaps/location/http.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  SnapManifest,
+  assertIsSnapManifest,
+  VFile,
+  HttpSnapIdStruct,
+} from '@metamask/snaps-utils';
+import { assert, assertStruct } from '@metamask/utils';
+
+import { ensureRelative } from '../../utils';
+import { SnapLocation } from './location';
+
+export default class HttpLocation implements SnapLocation {
+  // We keep contents separate because then we can use only one Blob in cache,
+  // which we convert to Uint8Array when actually returning the file.
+  //
+  // That avoids deepCloning file contents.
+  // I imagine ArrayBuffers are copy-on-write optimized, meaning
+  // in most often case we'll only have one file contents in common case.
+  private readonly cache = new Map<string, { file: VFile; contents: Blob }>();
+
+  private validatedManifest?: VFile<SnapManifest>;
+
+  private readonly url: URL;
+
+  constructor(url: URL) {
+    assertStruct(url.toString(), HttpSnapIdStruct, 'Invalid Snap Id: ');
+    this.url = url;
+  }
+
+  async manifest(): Promise<VFile<SnapManifest>> {
+    if (this.validatedManifest) {
+      return this.validatedManifest.clone();
+    }
+
+    // jest-fetch-mock doesn't handle new URL(), we need to convert this.url.toString()
+    const contents = await (await fetch(this.url.toString())).text();
```

Comment:
I don't see a reason for this. Actually we should remove fetch from `NpmLocation`.

`fetchMock` does the job for testing external fetches. And `SnapController` gets mocked locations.

Reply:
Hmm, I thought it would be good to have it parameterized so other environments could swap out fetch for whatever they need. If we have it in `NpmLocation` I think we should have it for the other locations too